_posts/2024-05-28-edb-and-kasten.markdown

```diff
@@ -0,0 +1,295 @@
+date: 2024-05-28 13:48:00 +0000
+author: michaelcourcy
+image: '/images/posts/2024-05-28-edb-and-kasten/kasten-edb-2.jpg'
+image_caption: 'EDB and Kasten partnership'
+tags: [Disaster Recovery, Migration, EDB, Kasten, RTO, Partnership, Managed Database]
+featured:
+---
+
+# Why database on Kubernetes ?
+
+ Having your database in Kubernetes has several advantages :   
+- **Performance** : you colocalize the data and the application
+- **Security** : All stay within the kubernetes network
+- **Ease of deployment** : `kubectl apply -f ...` and you're good to go. 
+- **Automation** : You define your desired state, the operator make it happens.
+- **Self Healing** : As it benefit to any workload in Kubernetes.
+ 
+However there is challenges : 
+- **Enterprise scale** : HA and big volumes are required.
+- **Efficient Protection** : Incremental backup, transaction consistency.
+- **Ease of use** : GUI and easy to use API. 
+- **Skill shortage** : Database and kubernetes double skills are hard to find.
+- **Migration and replication**  : we are more than ever in a hybrid momentum.
+
+Especially the last one **Migration and replication** :  no serious player let himself being trapped in a single cloud provider. How do you 
+negociate your bill if you cannot easily move ? How do you adapt to any change in the legal compliance of your country ?
+How do you move and replicate accross the cloud and the datacenters ? 
+ 
+EDB and Kasten together can overcome all these challenges let's see how. 
+ 
+# How EDB and Kasten work together ? 
+
+## EDB on Kubernetes 
+
+[EDB](https://www.enterprisedb.com) offer a full management of postgres clusters on Kubernetes. The only thing you have to do is create a custom resource like this one :
+
+```
+apiVersion: postgresql.k8s.enterprisedb.io/v1
+kind: Cluster
+metadata:
+  name: cluster-example
+  annotations:
+    "k8s.enterprisedb.io/addons": '["kasten"]'    
+spec:
+  instances: 3
+  storage:
+    size: 1Gi
+ ```
+ 
+"Et Voila", you immediately get a postgres cluster with one primary and 2 replicas. If the primary fail then the failover to one of the replica is immediate. Not only the operator will handle the failover it will also take care of repairing by creating another replicas so that you do not stay 
+in a degraded situation. All that automatically.
+ 
+EDB is the company behind the very well known Open Source Kubernetes project [CloudNativePG](https://cloudnative-pg.io), and you can get support and professional services from EDB through their [commercial offering](https://www.enterprisedb.com/services-support/professional-services). You can see 
+an EDB database as a managed database but without being tied in a cloud provider or a data center.
+
+
+## Solution 
+
+In it's regular run EDB manage the postgres cluster and make sure one instance is primary (read-write) and 2 replicas (read-only) are available in 
+case of failure to do the failover.
```

Comment:
In it's regular run EDB manages the postgres cluster ensuring there is always one primary (read-write) and 2 replicas (read-only) to handle failover scenarios.